src/actions/huntprey.ts

```
@@ -1,105 +1,109 @@
-import { ChatMessagePF2e } from "foundry-pf2e";
-
-export async function checkForHuntPreyGM(chatMessage: ChatMessagePF2e) {
-    if (!chatMessage.flags?.pf2e?.origin?.rollOptions?.includes("origin:item:hunt-prey")) return;
-
-    const ranger = game.actors.get(chatMessage.speaker.actor);
-    const rangerTargets = chatMessage.author?.targets;
-
-    if (!rangerTargets || rangerTargets.size === 0) {
-        return;
-    }
-
-    const allowedNumberOfTargets = await calculateNumberOfTargetsRangerCanHunt(ranger);
-    if (rangerTargets.size > allowedNumberOfTargets) {
-        return;
+import { ActorPF2e, ChatMessagePF2e, ScenePF2e, TokenDocumentPF2e } from "foundry-pf2e";
+import { ImageFilePath } from "foundry-pf2e/foundry/common/constants.mjs";
+import { AUTO_HUNT_PREY, getSocket } from "../sockets.ts";
+
+export async function runHuntPreyAsGM(rangerUuid: string, targetUuids: string[]) {
+
+    const ranger = fromUuidSync(rangerUuid) as ActorPF2e;
```

For consistency, `fromUuidSync<ActorPF2e>(rangerUuid)`

---

I've done this but it actually then seems to complain about `ranger` on lines 16 and 17. Is it doing something marginally different?

---

It's because `fromUuidSync` might not find anything, so it could be null.

By using `as ActorPF2e`, you're saying it's not null, so you're hiding that possible error.